src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx

```diff
@@ -59,8 +59,8 @@ function MentionUserRenderer({style, tnode, TDefaultRenderer, currentUserPersona
     if (!isEmpty(htmlAttribAccountID)) {
         const user = personalDetails[htmlAttribAccountID];
         accountID = parseInt(htmlAttribAccountID, 10);
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        displayNameOrLogin = PersonalDetailsUtils.getDisplayNameOrDefault(user, LocalePhoneNumber.formatPhoneNumber(user?.login ?? ''));
+        displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(user?.login ?? '') || PersonalDetailsUtils.getDisplayNameOrDefault(user);
+        displayNameOrLogin = Str.removeSMSDomain(getMentionDisplayText(displayNameOrLogin, htmlAttributeAccountID, user?.login ?? ''));
```

Comment:
This seems good, but I'm a little confused about what `getMentionDisplayText` is doing here. Can we clean it up too? 

For example, it doesn't look like we ever actually pass the third argument (`userLogin`) to `getMentionDisplayText`.

Reply:
https://github.com/Expensify/App/blob/6571b37784739d849aa36eddcf5301c08908371e/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L43-L57
We add `getMentionDisplayText` here to handle shortened mentions, and this is its job on email-based mention, so we decide to reuse it here rather than duplicate its code.

>  it doesn't look like we ever actually pass the third argument (`userLogin`).

We actually pass it in this PR to return early if `displayText` is not an email, but we ever pass it before because we don't need it for email-based mention, as it important only for  accountID-based mention.

And about clean `getMentionDisplayText`, this function job is to handle shortened mentions so we can rename it to `returnShortMentionIfFound`, and the `return early` inside it is related to accountID-based so we can move to its code logic.
```javascript
// replace `getMentionDisplayText` with function do its job.
const returnShortMentionIfFound = (userLogin) => {
    // If the emails are not in the same private domain, we also return the displayText
    if (!LoginUtils.areEmailsFromSamePrivateDomain(userLogin, currentUserPersonalDetails.login ?? '')) {
        return userLogin;
    }

    // Otherwise, the emails must be of the same private domain, so we should remove the domain part
    return userLogin.split('@')[0];
};

// accountID-based code will be
if (user?.login) {
    displayNameOrLogin = Str.isSMSLogin(user?.login) ? LocalePhoneNumber.formatPhoneNumber(user?.login) : returnShortMentionIfFound(user?.login)
}else{
    displayNameOrLogin = PersonalDetailsUtils.getDisplayNameOrDefault(user);
}
        
// in email-based mention replace `getMentionDisplayText` with `returnShortMentionIfFound`
```
This will make the code more clean but require a bit of changes more than the current shorthand.
What do you think @puneetlath ?

Reply:
I like that approach. 

So then we could do something like: 
```
displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(user?.login ?? '') || PersonalDetailsUtils.getDisplayNameOrDefault(user);
mentionDisplayText = Str.removeSMSDomain(getShortMentionIfFound(displayNameOrLogin, htmlAttributeAccountID, user?.login ?? ''));


What do you think @sobitneupane?

Reply:
@puneetlath I am okay with renaming the function's and variable's name.

For the code that you are suggesting [here](https://github.com/Expensify/App/pull/38957#discussion_r1541380980), I think we will need the current implementation of `getShortMentionIfFound`, right @ahmedGaber93 ?

For the code that @ahmedGaber93 has [suggested](https://github.com/Expensify/App/pull/38957#discussion_r1539958572), we need to first determine whether the `displayNameorLogin` is login or not and call the `getShortMentionIfFound` function only if `displayNameorLogin` is login.

IMO, the better idea is to let `getShortMentionIfFound` to handle the logic completely(current implementation).

Reply:
Ok sounds good. My main point is that if we can consolidate/clean up some of this logic while we are here, I think we should.

Reply:
Updated!
@sobitneupane @puneetlath

Reply:
Thanks for the update @ahmedGaber93